core/deployment/src/main/java/io/quarkus/deployment/pkg/steps/GraalVM.java

```diff
@@ -8,24 +8,28 @@
 final class GraalVM {
     static final class Version implements Comparable<Version> {
         private static final Pattern PATTERN = Pattern.compile(
-                "(GraalVM|native-image)( Version)? ([1-9][0-9]*)\\.([0-9]+)\\.[0-9]+(-dev\\p{XDigit}*)?([^\n$]*)\\s*");
+                "(GraalVM|native-image)( Version)? (?<version>[1-9][0-9]*(\\.[0-9]+)+(-dev\\p{XDigit}*)?)(?<distro>[^\n$]*)(Java Version (?<javaVersion>[^)]+))?\\s*");
 
-        static final Version UNVERSIONED = new Version("Undefined", -1, -1, Distribution.ORACLE);
-        static final Version VERSION_21_2 = new Version("GraalVM 21.2", 21, 2, Distribution.ORACLE);
-        static final Version VERSION_21_3 = new Version("GraalVM 21.3", 21, 3, Distribution.ORACLE);
+        static final Version UNVERSIONED = new Version("Undefined", "snapshot", Distribution.ORACLE);
+        static final Version VERSION_21_2 = new Version("GraalVM 21.2", "21.2", Distribution.ORACLE);
+        static final Version VERSION_21_3_0 = new Version("GraalVM 21.3.0", "21.3.0", Distribution.ORACLE);
 
         static final Version MINIMUM = VERSION_21_2;
-        static final Version CURRENT = VERSION_21_3;
+        static final Version CURRENT = VERSION_21_3_0;
```

Comment:
I'm not sure we want that? IIRC, we use `CURRENT` in some error messages and I'm pretty sure we prefer advertising the minor and not the micro. Or am I mistaken?

If I were you, I would have kept `VERSION_21_3`, kept it as the current version and introduce `VERSION_21_3_0` as a brand new constant.

Reply:
> I'm not sure we want that? IIRC, we use `CURRENT` in some error messages and I'm pretty sure we prefer advertising the minor and not the micro. Or am I mistaken?

You are right, thanks!